.drone.star

```diff
@@ -527,8 +547,33 @@ def coreApiTests(ctx, part_number = 1, number_of_parts = 1, storage = "owncloud"
         "volumes": [pipelineVolumeOC10Tests],
     }
 
+def apiTests(ctx, pipelines):
+    debugParts = config["apiTests"]["skipExceptParts"]
+    for runPart in range(1, config["apiTests"]["numberOfParts"] + 1):
+        if len(debugParts) != 0:
+            if runPart in debugParts:
+                pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "owncloud"))
+                pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "ocis"))
+        else:
+            pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "owncloud"))
+            pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "ocis"))
```

Comment:
```suggestion
    debugPartsEnabled = (len(debugParts) != 0)
    for runPart in range(1, config["apiTests"]["numberOfParts"] + 1):
        if (!debugPartsEnabled or (debugPartsEnabled and runPart in debugParts):
                pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "owncloud"))
                pipelines.append(coreApiTests(ctx, runPart, config["apiTests"]["numberOfParts"], "ocis"))
```
We could avoid repeating the `pipelines.append` lines by condensing the logic - something like this.